packages/console/src/pages/UserDetails/UserRoles/index.tsx

```diff
@@ -131,6 +132,17 @@ const UserRoles = () => {
           {t('user_details.roles.delete_description')}
         </ConfirmModal>
       )}
+      {isAssignRolesModalOpen && (
+        <AssignRolesModal
+          user={user}
+          onClose={(success) => {
+            if (success) {
+              void mutate();
+            }
+            setIsAssignRolesModalOpen(false);
+          }}
+        />
+      )}
```

Comment:
i think the modal logic with a state to control open/close is common, should we wrap it into a hook?

Reply:
if so, no need to do it now

Reply:
Add an issue to track this:
https://linear.app/silverhand/issue/LOG-5163/align-modal-open-logic